internal/commands/database/delete_test.go

```diff
@@ -0,0 +1,53 @@
+package database
+
+import (
+	"testing"
+
+	"github.com/UpCloudLtd/upcloud-cli/internal/commands"
+	"github.com/UpCloudLtd/upcloud-cli/internal/config"
+	smock "github.com/UpCloudLtd/upcloud-cli/internal/mock"
+
+	"github.com/UpCloudLtd/upcloud-go-api/v4/upcloud"
+	"github.com/UpCloudLtd/upcloud-go-api/v4/upcloud/request"
+	"github.com/gemalto/flume"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestDeleteCommand(t *testing.T) {
+	targetMethod := "DeleteManagedDatabase"
+
+	database := upcloud.ManagedDatabase{
+		UUID: "27fbd082-30b0-11eb-adc1-0242ac120004",
+		Name: "test-database",
+	}
+
+	for _, test := range []struct {
+		name  string
+		arg   string
+		error string
+		req   request.DeleteManagedDatabaseRequest
+	}{
+		{
+			name: "delete with UUID",
+			arg:  database.UUID,
+			req:  request.DeleteManagedDatabaseRequest{UUID: database.UUID},
+		},
+	} {
+		t.Run(test.name, func(t *testing.T) {
+			mService := smock.Service{}
+			mService.On(targetMethod, &test.req).Return(nil)
+
+			conf := config.New()
+			c := commands.BuildCommand(DeleteCommand(), nil, conf)
+
+			_, err := c.(commands.MultipleArgumentCommand).Execute(commands.NewExecutor(conf, &mService, flume.New("test")), test.arg)
```

Comment:
Alternatively, [mockexecute](https://github.com/UpCloudLtd/upcloud-cli/blob/main/internal/mockexecute/mockexecute.go) helper could be used to simulate the command execution, but as it mostly helps testing Cobras argument validations and checking output, it would not bring much benefits here.